geopandas/tools/sjoin.py

```diff
@@ -84,6 +94,33 @@ def sjoin(
     Every operation in GeoPandas is planar, i.e. the potential third
     dimension is not taken into account.
     """
+    if "op" in kwargs:
+        op = kwargs.pop("op")
+        deprectation_message = (
+            "The `op` parameter is deprecated and will be removed"
+            " in a future release. Please use the `predicate` parameter"
+            " instead."
+        )
+        if predicate != "intersects" and op != predicate:
+            override_message = (
+                "A non-default value for `predicate` was passed"
+                f' (got `predicate="{predicate}"`'
+                f' in combination with `op` (got `op="{op}"`).'
+                " The value of `predicate` will be overriden by the value of `op`,"
+                " , which may result in unexpected behavior."
+                f"\n{deprectation_message}"
+            )
+            warnings.warn(override_message, UserWarning, stacklevel=4)
+        else:
+            warnings.warn(deprectation_message, FutureWarning, stacklevel=4)
+    if kwargs:
```

Comment:
Would be good to model this off other functions, e.g.,  

```
>>> df.contains(df.geometry, foo='bar')
...
TypeError: contains() got an unexpected keyword argument 'foo'
```

Reply:
makes sense, simpler as well